bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/Debouncer.java

```diff
@@ -74,7 +76,14 @@ void call() {
         lastCallAttempt = clock.millis();
         calls.incrementAndGet();
         if (pending.compareAndSet(false, true)) {
-            scheduler.schedule(this::tryActionOrPostpone, delayMs, TimeUnit.MILLISECONDS);
+            feature = scheduler.schedule(this::tryActionOrPostpone, delayMs, TimeUnit.MILLISECONDS);
```

Comment:
Instead of this constantly rescheduling itself why not just have it called with a fixed delay. 
Then you can have the future canceled inside of `tryActionOrPostpone` if it doesn't want to be called again?

Reply:
im just afraid to touch this working logic. it must be a reason why it was done in this rather complex way.

Reply:
Yeah, you are right, it is doing some funky logic. Doesn't look like it can really be changed either. After looking more closely I actually kinda like the way it was designed, so yeah, don't change it.